src/examples/CapiResize.cpp

```diff
@@ -0,0 +1,125 @@
+#include <Elementary.h>
+#include <thorvg_capi.h>
+
+#include <stdlib.h>
+
+#define WIDTH 400
+#define HEIGHT 400
+
+
+/************************************************************************/
+/* Capi Test Code                                                       */
+/************************************************************************/
+
+static uint32_t *buffer = NULL;
+
+static Tvg_Paint *shape = NULL;
+static Tvg_Paint *scene = NULL;
+
+static Tvg_Canvas *canvas = NULL;
+
+static Eo *view = NULL;
+
+void init()
+{
+    canvas = tvg_swcanvas_create();
+    scene = tvg_scene_new();
+    shape = tvg_shape_new();
+
+    tvg_paint_set_opacity(shape, 0);
+    tvg_shape_append_rect(shape, 100, 0, 100, 100, 0, 0);
+    tvg_shape_set_stroke_color(shape, 255, 0, 0, 255);
+    tvg_shape_set_stroke_width(shape, 1);
+
+    buffer = (uint32_t*) malloc(sizeof(uint32_t) * WIDTH * HEIGHT);
+    tvg_swcanvas_set_target(canvas, buffer, WIDTH, WIDTH, HEIGHT, TVG_COLORSPACE_ARGB8888);
+
+    tvg_canvas_clear(canvas, false);
+    tvg_scene_clear(scene);
+    tvg_canvas_push(canvas, scene);
+
+    tvg_paint_translate(shape, 0, 0);
+    tvg_shape_set_fill_color(shape, 0, 0, 0, 0);
+    tvg_scene_push(scene, shape);
+    tvg_paint_set_opacity(shape, 255);
+
+    tvg_canvas_update(canvas);
+    tvg_canvas_draw(canvas);
+    tvg_canvas_sync(canvas);
+}
+
+static void _resize_cb(void *data, Evas *e, Evas_Object *obj, void *event_info)
+{
+    int w = 0, h = 0;
+    evas_object_geometry_get(obj, NULL, NULL, &w, &h);
+
+    if ((w != WIDTH || h != HEIGHT) && (w != 0 && h != 0))
+    {
+      buffer = (uint32_t*) realloc(buffer, sizeof(uint32_t) * w * h);
+
+      tvg_swcanvas_set_target(canvas, buffer, w, w, h, TVG_COLORSPACE_ARGB8888);
+
+      tvg_canvas_clear(canvas, false);
+      tvg_scene_clear(scene);
+
+      tvg_canvas_push(canvas, scene);

```

Comment:
scene is already pushed in the canvas...

Reply:
Can we merge this _resize_cb to Capi.cpp? (to not add new file)

You don't need to do nothing for paints here but just call reset target & redraw to guarantee the issue.

static void _resize_cb(..)
{
    int w = 0, h = 0;
    evas_object_geometry_get(obj, NULL, NULL, &w, &h);

    if ((w != WIDTH || h != HEIGHT) && (w != 0 && h != 0))
    {
      buffer = (uint32_t*) realloc(buffer, sizeof(uint32_t) * w * h);
      tvg_swcanvas_set_target(...);
      tvg_swcanvas_update(canvas);
      tvg_swcanvas_draw();
      tvg_swcanvas_sync();
      ...
}